### PR TITLE
Adding telemetry points to connection dialog

### PR DESCRIPTION
**Pull request by Benjin**
Addresses #18261

* dialog initialization failure
* connection option construction failure
* loading connections from MRU list
* connection success/failure
* azure browsing success/failure

### CODE REVIEW

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 1]
```diff
@@ -119,9 +124,15 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
...
         this.registerRpcHandlers();
-        this.initializeDialog().catch((err) =>
-            vscode.window.showErrorMessage(getErrorMessage(err)),
-        );
+        this.initializeDialog().catch((err) => {
+            void vscode.window.showErrorMessage(getErrorMessage(err));
+            sendErrorEvent(
+                TelemetryViews.ConnectionDialog,
+                TelemetryActions.Initialize,
+                err,
```
Comment (aasimkhan30) [inline src/connectionconfig/connectionDialogWebviewController.ts:132]: No potential PIIs?

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 2]
```diff
@@ -921,18 +942,71 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
...
                         return state;
                     }
                 } catch (error) {
                     this.state.formError = getErrorMessage(error);
                     this.state.connectionStatus = ApiStatus.Error;
+
+                    sendErrorEvent(
+                        TelemetryViews.ConnectionDialog,
+                        TelemetryActions.CreateConnection,
+                        error,
```
Comment (aasimkhan30) [inline src/connectionconfig/connectionDialogWebviewController.ts:982]: PIIs?

**Review by aasimkhan30 — APPROVED** [review]
Looks good to me. Please make sure that error messages don't have any PIIs.

File: src/connectionconfig/connectionDialogWebviewController.ts (hunk 2 above)
Comment (Benjin) [reply src/connectionconfig/connectionDialogWebviewController.ts:982]: `includeErrorMessage` defaults to `false`, so I think this'll be okay, but I'll explicitly set that to be sure

File: src/connectionconfig/connectionDialogWebviewController.ts (hunk 1 above)
Comment (Benjin) [reply src/connectionconfig/connectionDialogWebviewController.ts:132]: The spots in `initializeDialog()` that handle potential PII have their own error catches that emit error telemetry, and those spots have `includeErrorMessage` set to `false`.  Everything else during initialization shouldn't have PII.

**Comment by aasimkhan30** [comment]
Synced up offline and the events seem to be free of any PII

**Pull request merged**